### DISCUSSION TITLE
Theme Support

### DISCUSSION THREAD

**Discussion by mkwatchface** [agronick/aa-torque#58] [Ideas]
Hello,

This is a great app! I tested today and it works flawlessly!!! I would like to create more themes for the app but I wouldn't know where to start.

I can see that there is test theme in the options that seem to be the proportions needed for the dial but other than that I am a bit lost...

Is there a guide on creating themes? Interested in: what is the resolution, does it support fonts, how should it be named, how to test a new theme, where can I send it if I manage to make one?

It would be great if the app had a "Browse Theme" button that could give you the ability to load any theme from the filesystem (a folder containing the theme)...

Thank you very much
Kind Regards

**Comment by agronick** [comment]
It would be a lot of work to import themes. Someone else is welcome to add that functionality. All the themes are defined here https://github.com/agronick/aa-torque/blob/master/app/src/main/res/values/styles.xml They are regular android themes. You can find more about it by looking at theming on android.

**Comment by mkwatchface** [comment]
Thank you for your reply!
Unfortunately, i don't have the knowledge to do it myself. I created a theme today based on the images i found in the res folder. It looks like this:

![Preview](https://github.com/agronick/aa-torque/assets/62142158/1688a08b-9ac5-4845-a442-d8afb29f4031)

Could i send it to you to be included in the next update?
How can i send it to you?

Thank you

**Reply by agronick** [reply]
Lookks good. You can attach it here. You need the background and the needle in separate images using the dimensions as the existing images.

**Reply by mkwatchface** [reply]
Yeah I have it ready in separate images. I will also attach the psd so if you have photoshop you can edit anything if needed. If you don't have photoshop you can still open the psd with https://www.photopea.com/...

[Skoda Virtual Cockpit.zip](https://github.com/agronick/aa-torque/files/13196430/Skoda.Virtual.Cockpit.zip)


Hope this helps

I can design more if needed!

**Reply by mkwatchface** [reply]
I would like to inform you that i edited the graphics as i found a tiny error. Please download again!

Sorry for the inconvenience!

Kind Regards
Michail Kritikos

**Reply by agronick** [reply]
Ok, I'll need to find some time to integrate it and test it. I'm working on some pretty big features right now while also trying to find a job. You can fork the repo and try changing it yourself by just replacing the images in an existing theme. It should compile on github.

**Reply by mkwatchface** [reply]
Hi sorry to bother again,

i have tried to fork the repo, added the images in the drawable folder, updated the values/values.xml and tried to release, however, github only creates a source code zip and tar.gz file and does not compile to apk.

As i mentioned before, i do not have the knowledge to do it on my own :( (i work better with graphics instead of code) so i will wait for an official update containing the dial i built, if of course you choose to include it!

This is more or less what it would look like:

![Preview in car](https://github.com/agronick/aa-torque/assets/62142158/82e5132e-60f0-43ba-91fd-9a785e3c651c)
![278722956-e31b0598-25ec-4003-bcee-11e5e108d3a4 copy](https://github.com/agronick/aa-torque/assets/62142158/f335c050-1922-4bf6-8f1d-2226bcae705b)


Thank you for your efforts and wish you the best of finding a new job!
Kind Regards

**Reply by agronick** [reply]
Looks nice. Do you have a name for it. When I do it I'll put it in a separate commit so you can see what was done if you want to do it on other ones.

Actually, it looks like it is building releases for you. I'll look at your code. https://github.com/mkwatchface/aa-torque/releases

**Reply by mkwatchface** [reply]
Thank you, it looks nice. It was built based on the virtual cockpit of my skoda scala monte carlo. So we can name it "Skoda Monte Carlo".

Yes, github is building the release but it's only the source code. I was expecting it would compile to apk so I could install...

**Reply by agronick** [reply]
I made a PR for your changes and made a guide on how to create themes. https://github.com/agronick/aa-torque/pull/63 Can you add me as a maintainer on your fork so I can see what is going on with the actions.

**Reply by mkwatchface** [reply]
Thank you for the guide!!! that's awesome... let me rework it because based on the guide i miss one image... Will post here of help is needed!

**Reply by agronick** [reply]
I posted your theme in a prerelease. If everything goes good I'll probably switch it to a full release tonight. If you come up with a name for it let me know.

**Reply by mkwatchface** [reply]
Thanks a million!!! I think we can call it "Skoda Monte Carlo"

**Reply by mkwatchface** [reply]
Just installed it... I think I have the images named opposite... when I choose to show ticks they are not visible, and I choose not to show them they are there... please fix before the final release!!!

**Reply by agronick** [reply]
I think its the xml that was backwards. It looks really good. I'll probably make this my daily driver. I think the needle should be bigger though. It looks really small on my screen.  https://photos.app.goo.gl/fFYxQ3cEAaZzAaRF6

**Reply by agronick** [reply]
So I just pushed up a commit that fixes it. I also turned on tickRotation since it matched your inspiration but there was no good way to put the ticks right on the dial and have it work for all screen sizes. So the numbers are rotated inside the dial. If you want to change any of it let me know.

**Reply by mkwatchface** [reply]
It looks good! But i will make the needle a bit thicker so it is more visible.
I will attach here as soon as it's done, just give me 2-3 minutes

**Reply by mkwatchface** [reply]
Here is the needle a bit thicker (from 3 pixels to 7 pixels). I also added a slight shadow so it separates more from the background.

![needle_mk](https://github.com/agronick/aa-torque/assets/62142158/2d80be9a-f2f9-4a38-82e4-6d99ffbcd343)

**Reply by mkwatchface** [reply]
BTW for the Monte Carlo, let's not have the ticks rotated please... if you have a small screen and you add rpm to thousands the numbers are touching each other.

Let's disable the tickrotation

Thank you!!!

**Reply by Vince56190** [reply]
I love it , is it possible to create the same in blue and white Needle . I don't have the skills to do that

**Reply by mkwatchface** [reply]
Hello,

So, blue outer ring, blue ticks, white needle, but keep the red area on high value right? I mean if the dial is rpm from 6K to 8K RPM it would remain red?

Also, the blue should be light blue or dark blue?

**Reply by Vince56190** [reply]
All the red parts in Dark blue exept needle in white , no red zone just a shade of blue thanks
I ask to my follower ( Chrysler 300 group) to design jauge like OEM one 
![PXL_20230401_114022151](https://github.com/agronick/aa-torque/assets/147246549/6f8e115b-8f0a-4667-b311-f7484aa93041)

![IMG_20200801_105536](https://github.com/agronick/aa-torque/assets/147246549/a9a458d4-5723-4bea-b4e3-d785ff5d21f5)

**Reply by mkwatchface** [reply]
Kyle hi, please when i install the latest pre-release i can open it on the phone but when i try to force update i get the bellow error: ![Screenshot_20231101_174658_Package installer](https://github.com/agronick/aa-torque/assets/62142158/8b7ee449-f2eb-43a3-9716-d77f3e8824bd)

Is this because it is a pre-release?

**Reply by Vince56190** [reply]
i dont have this issue ( Android 13 , Android Auto beta 10.8 and AATorque 14.1) i installed it with kinginstaller
![Screenshot_20231101_182332_AA Torque](https://github.com/agronick/aa-torque/assets/147246549/e51be033-6b59-49d3-8950-65fb0b46415b)

**Reply by agronick** [reply]
I made it a final release. The new needle looks really good.

**Reply by mkwatchface** [reply]
Hello again! Thanks a million! I have updated it on the phone but i didn't have the chance to drive my car with it! On the phone it looks amazing. I would like to be able to add themes myself and i try to follow your guide which is really good! I am able to add the images and change/add the XML code but on the repo the build_deply always fails. I have sent you an invitation to the repo, but in any case, if anyone else is able to help i would much appreciate it as i can understand that you can be busy!

**Reply by Vince56190** [reply]
It look well on a VW transporter https://photos.app.goo.gl/nkGicQQdGFCaR8aC9

**Reply by agronick** [reply]
Ok, I fixed it so you should be able to create releases. I also made it so the version.properties file wouldn't be changed since it would overwrite the main repo when it is merged.

**Reply by mkwatchface** [reply]
YEEEES... Now it works! Unfortunately until now i didn't have the opportunity to test it on my phone so on the latest Theme i created the ticks are very close to the center. The padding needs to change to have the numbers closer to the frame of the dial!

Thank you!!!

**Reply by agronick** [reply]
Yeah I adjusted the ticks but it looks good. I haven't gotten a chance to test it in the car either. It will be cool to see how the needle moves with the body of the diial spinning. It might loo good if you bring down the alpha channel a little bit on the black parts as well.

**Comment by AnDr3w9147** [comment]
I will say this app is magnificent, thank you for all your hard work!!!

**Comment by mkwatchface** [comment]
I already started to design a new one:

![Dial](https://github.com/agronick/aa-torque/assets/62142158/bc9cd57f-cd77-4751-bdf9-502a42549e71)

**Reply by ghynett** [reply]
Very good! Can you create a Citroen theme coherent with the original car theme? In attachment the shot of the car cockpit.
![20231125_113705](https://github.com/agronick/aa-torque/assets/60058899/6d88c185-33ba-4282-9a57-9160eb7af90d)
